<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MapEditorClass</class>
 <widget class="QMainWindow" name="MapEditorClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1060</width>
    <height>691</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MapEditor</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QStackedWidget" name="stackedWidget">
    <property name="geometry">
     <rect>
      <x>-30</x>
      <y>0</y>
      <width>1091</width>
      <height>641</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>2</number>
    </property>
    <widget class="QWidget" name="widget_mainPage">
     <widget class="QLabel" name="label_mainPage_title">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>50</y>
        <width>581</width>
        <height>241</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Goudy Stout</family>
        <pointsize>36</pointsize>
        <underline>true</underline>
       </font>
      </property>
      <property name="text">
       <string>Map Editor</string>
      </property>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>300</x>
        <y>210</y>
        <width>441</width>
        <height>351</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="layout_mainMenu">
       <item>
        <widget class="QPushButton" name="button_mainPage_newMap">
         <property name="font">
          <font>
           <family>Goudy Stout</family>
           <pointsize>16</pointsize>
          </font>
         </property>
         <property name="text">
          <string>New Map</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="button_mainPage_loadMap">
         <property name="font">
          <font>
           <family>Goudy Stout</family>
           <pointsize>16</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Load Map</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="button_mainPage_exit">
         <property name="font">
          <font>
           <family>Goudy Stout</family>
           <pointsize>16</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Exit</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="widget_createNewMap">
     <widget class="QPushButton" name="button_createNewMap_createMap">
      <property name="geometry">
       <rect>
        <x>410</x>
        <y>410</y>
        <width>221</width>
        <height>71</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Goudy Stout</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Start</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_createNewMap_title">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>20</y>
        <width>411</width>
        <height>241</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Goudy Stout</family>
        <pointsize>36</pointsize>
        <underline>true</underline>
       </font>
      </property>
      <property name="text">
       <string>New map</string>
      </property>
     </widget>
     <widget class="QGroupBox" name="newMapInput">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>190</y>
        <width>571</width>
        <height>191</height>
       </rect>
      </property>
      <property name="title">
       <string/>
      </property>
      <widget class="QLineEdit" name="input_createNewMap_mapName">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>40</y>
         <width>391</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_createNewMap_nameInput">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>10</y>
         <width>131</width>
         <height>21</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Goudy Stout</family>
        </font>
       </property>
       <property name="text">
        <string>Map Name</string>
       </property>
      </widget>
      <widget class="QSplitter" name="splitter">
       <property name="geometry">
        <rect>
         <x>340</x>
         <y>90</y>
         <width>145</width>
         <height>44</height>
        </rect>
       </property>
       <property name="orientation">
        <enum>Qt::Orientation::Vertical</enum>
       </property>
       <widget class="QLabel" name="label_createNewMap_mapHeight">
        <property name="font">
         <font>
          <family>Goudy Stout</family>
         </font>
        </property>
        <property name="text">
         <string>Map Height</string>
        </property>
       </widget>
       <widget class="QSpinBox" name="input_createNewMap_mapHeight">
        <property name="font">
         <font>
          <family>Goudy Stout</family>
          <pointsize>12</pointsize>
         </font>
        </property>
       </widget>
      </widget>
      <widget class="QSplitter" name="splitter_2">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>90</y>
         <width>136</width>
         <height>44</height>
        </rect>
       </property>
       <property name="orientation">
        <enum>Qt::Orientation::Vertical</enum>
       </property>
       <widget class="QLabel" name="label_createNewMap_mapWidth">
        <property name="font">
         <font>
          <family>Goudy Stout</family>
         </font>
        </property>
        <property name="text">
         <string>Map Width</string>
        </property>
       </widget>
       <widget class="QSpinBox" name="input_createNewMap_mapWidth">
        <property name="font">
         <font>
          <family>Goudy Stout</family>
          <pointsize>12</pointsize>
         </font>
        </property>
       </widget>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="widget_loadMap">
     <widget class="QPushButton" name="button_loadMap_load">
      <property name="geometry">
       <rect>
        <x>700</x>
        <y>220</y>
        <width>201</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Load Map</string>
      </property>
     </widget>
     <widget class="QListView" name="list_loadMap_mapList">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>80</y>
        <width>361</width>
        <height>511</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Goudy Stout</family>
        <pointsize>12</pointsize>
       </font>
      </property>
     </widget>
     <widget class="QLabel" name="label_loadMap_title">
      <property name="geometry">
       <rect>
        <x>380</x>
        <y>20</y>
        <width>561</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Load Map Menu</string>
      </property>
     </widget>
     <widget class="QPushButton" name="button_loadMap_exit">
      <property name="geometry">
       <rect>
        <x>700</x>
        <y>280</y>
        <width>201</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Exit</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="widget_mapEditor">
     <widget class="QLabel" name="label_mapEditor_title">
      <property name="geometry">
       <rect>
        <x>430</x>
        <y>20</y>
        <width>251</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Goudy Stout</family>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Map Editor</string>
      </property>
     </widget>
     <widget class="QPushButton" name="button_mapEditor_reset">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>10</y>
        <width>141</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Reset</string>
      </property>
     </widget>
     <widget class="QPushButton" name="button_mapEditor_save">
      <property name="geometry">
       <rect>
        <x>920</x>
        <y>470</y>
        <width>131</width>
        <height>61</height>
       </rect>
      </property>
      <property name="text">
       <string>Save</string>
      </property>
     </widget>
     <widget class="QPushButton" name="button_mapEditor_exit">
      <property name="geometry">
       <rect>
        <x>920</x>
        <y>540</y>
        <width>131</width>
        <height>61</height>
       </rect>
      </property>
      <property name="text">
       <string>Exit</string>
      </property>
     </widget>
     <widget class="QGroupBox" name="box_position">
      <property name="geometry">
       <rect>
        <x>850</x>
        <y>180</y>
        <width>201</width>
        <height>241</height>
       </rect>
      </property>
      <property name="title">
       <string/>
      </property>
      <widget class="QSpinBox" name="input_mapEditor_endingColumn">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>190</y>
         <width>51</width>
         <height>41</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_mapEditor_startingRow">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>50</y>
         <width>49</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Row</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_mapEditor_endingRow">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>170</y>
         <width>49</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Row</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="input_mapEditor_endingRow">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>190</y>
         <width>51</width>
         <height>41</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_mapEditor_endingColumn">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>170</y>
         <width>61</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Column</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="input_mapEditor_startingColumn">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>70</y>
         <width>51</width>
         <height>41</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_mapEditor_endingPosition">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>140</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Ending position</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_mapEditor_startingPosition">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>20</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Starting position</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="input_mapEditor_startingRow">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>70</y>
         <width>51</width>
         <height>41</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_mapEditor_startingColumn">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>50</y>
         <width>61</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Column</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="box_mapEditor_placeWall">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>490</y>
        <width>241</width>
        <height>111</height>
       </rect>
      </property>
      <property name="title">
       <string>Place Wall</string>
      </property>
      <widget class="QLabel" name="label_mapEditor_placeWallRow">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>49</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Row</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="input_mapEditor_placeWallColumn">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>50</y>
         <width>51</width>
         <height>41</height>
        </rect>
       </property>
      </widget>
      <widget class="QSpinBox" name="input_mapEditor_placeWallRow">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>50</y>
         <width>51</width>
         <height>41</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_mapEditor_placeWallColumn">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>30</y>
         <width>61</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Column</string>
       </property>
      </widget>
      <widget class="QPushButton" name="button_mapEditor_placeWall">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>50</y>
         <width>81</width>
         <height>41</height>
        </rect>
       </property>
       <property name="text">
        <string>Place Wall</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="box_mapEditor_placeObject">
      <property name="geometry">
       <rect>
        <x>310</x>
        <y>490</y>
        <width>241</width>
        <height>111</height>
       </rect>
      </property>
      <property name="title">
       <string>Place Chest</string>
      </property>
      <widget class="QLabel" name="label_mapEditor_placeObjectRow">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>49</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Row</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="input_mapEditor_placeObjectColumn">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>50</y>
         <width>51</width>
         <height>41</height>
        </rect>
       </property>
      </widget>
      <widget class="QSpinBox" name="input_mapEditor_placeObjectRow">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>50</y>
         <width>51</width>
         <height>41</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_mapEditor_placeObjectColumn">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>30</y>
         <width>61</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Column</string>
       </property>
      </widget>
      <widget class="QPushButton" name="button_mapEditor_placeObject">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>50</y>
         <width>81</width>
         <height>41</height>
        </rect>
       </property>
       <property name="text">
        <string>Place Chest</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="box_mapEditor_emptyCell">
      <property name="geometry">
       <rect>
        <x>570</x>
        <y>490</y>
        <width>241</width>
        <height>111</height>
       </rect>
      </property>
      <property name="title">
       <string>Empty Cell</string>
      </property>
      <widget class="QLabel" name="label_mapEditor_emptyCellRow">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>49</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Row</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="input_mapEditor_emptyCellColumn">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>50</y>
         <width>51</width>
         <height>41</height>
        </rect>
       </property>
      </widget>
      <widget class="QSpinBox" name="input_mapEditor_emptyCellRow">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>50</y>
         <width>51</width>
         <height>41</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_mapEditor_emptyCellColumn">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>30</y>
         <width>61</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Column</string>
       </property>
      </widget>
      <widget class="QPushButton" name="button_mapEditor_emptyCell">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>50</y>
         <width>81</width>
         <height>41</height>
        </rect>
       </property>
       <property name="text">
        <string>Empty</string>
       </property>
      </widget>
     </widget>
     <widget class="GameMapWidget" name="widget_mapEditor_map" native="true">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>80</y>
        <width>741</width>
        <height>381</height>
       </rect>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1060</width>
     <height>33</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFunny_Button">
    <property name="title">
     <string>Funny Button</string>
    </property>
   </widget>
   <addaction name="menuFunny_Button"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GameMapWidget</class>
   <extends>QWidget</extends>
   <header>gamemapwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <includes>
  <include location="local">gamemapwidget.h</include>
 </includes>
 <resources/>
 <connections/>
</ui>
